### DISCUSSION TITLE
Installation in WSL (Windows Subsystem for Linux)

### DISCUSSION THREAD

**Discussion by JeffPlsFix** [manmay-nakhashi/tortoise-tts-fastest#10] [General]
## Has anyone made this work on WSL? I got it working:

https://gist.github.com/JeffPlsFix/9ff3137a42f6f1f5f8ba1a441eea413c

I wrote a Gist on how i installed this repo on WSL Ubuntu on Windows.
Setting up a WSL environment with Cuda Toolkit and Conda was a bit tricky to get right, but i think i got the installation sequence correct now.

I couldnt get the repo and its dependencies to function properly on Windows, so i went the Linx route only using WSL.
Ive also had some issues where pytorch would allocate excessive amounts of memory and throw OOM errors, which seemingly went away only when installing in this specific order. 💻 ❓ 

Please tell me about your experiences!

**Comment by kzhjymug** [comment]
Thanks for your guide. I feel it is the closest I've gotten to running any of the TTTS-fast options, but ran into a pytorch error, same one mentioned here: https://github.com/manmay-nakhashi/tortoise-tts-fastest/issues/12

I have experimented with a lot of ML projects like this over the past few years, I must say TTTS-fast has been the most frustrating to get working by far.

I have an RTX 3090, FWIW, so maybe there is some issue there, I used the 40-series instructions in your guide.

**Reply by JeffPlsFix** [reply]
Thanks! i am glad it helped you get some steps closer.

I believe there was some incompatability with something cuda 1.7 related and the 4000 series RTX cards.. Ill suggest you try the GTX 1000 series instructions since i believe only the RTX 4000 series and newer had problems that made the GTX 1000 instructions not work, so your RTX 3090 should be compatible with the older cuda versions.

I should probably rephrase it to "GTX 1000 to RTX 3000" and "RTX 4000 and newer" but i dont have an 3000 series card to test it with.

I know how frustrating it can be, i spent a lot of time figuring out the right procedure myself, but it mostly comes down to getting the correct versions of packages and cuda. If you get it working then give me an update!
